client/src/components/App/Content/InvestigationForm/TabManagement/ExposuresAndFlights/ExposuresAndFlights.tsx

```diff
@@ -1,37 +1,64 @@
-import { useSelector } from 'react-redux';
-import StoreStateType from 'redux/storeStateType';
-import { Collapse, Divider, Typography } from '@material-ui/core';
 import Swal from 'sweetalert2';
+import { useSelector } from 'react-redux';
+import { AddCircle } from '@material-ui/icons';
+import React, { useEffect, useContext, useState } from 'react';
+import { Collapse, Divider, Typography, IconButton } from '@material-ui/core';
 
 import axios from 'Utils/axios';
 import Toggle from 'commons/Toggle/Toggle';
+import StoreStateType from 'redux/storeStateType';
 import FormRowWithInput from 'commons/FormRowWithInput/FormRowWithInput';
-import { exposureAndFlightsContext, fieldsNames } from 'commons/Contexts/ExposuresAndFlights';
+import { exposureAndFlightsContext, fieldsNames, Exposure, initialExposureOrFlight, isConfirmedExposureInvalid, isFlightInvalid } from 'commons/Contexts/ExposuresAndFlights';
 
-import FlightsForm from './FlightsForm/FlightsForm';
-import ExposureForm from './ExposureForm/ExposureForm';
 import useFormStyles from 'styles/formStyles';
+import FlightsForm from './FlightsForm/FlightsForm';
 import useStyles from './ExposuresAndFlightsStyles';
+import ExposureForm from './ExposureForm/ExposureForm';
 
+const addConfirmedExposureButton: string = 'הוסף חשיפה';
+const addFlightButton: string = 'הוסף טיסה לחול';
 
 const ExposuresAndFlights = () => {
-  const context = useContext(exposureAndFlightsContext);
-  const { exposureAndFlightsData, setExposureDataAndFlights } = context;
+  const { exposureAndFlightsData, setExposureDataAndFlights } = useContext(exposureAndFlightsContext);;
+  const { exposures } = exposureAndFlightsData;
 
   const investigationId = useSelector<StoreStateType, number>((state) => state.investigation.epidemiologyNumber);
 
   const { fieldName } = useFormStyles();
   const classes = useStyles();
 
+  const disableConfirmedExposureAddition : boolean= React.useMemo(() => 
+    exposures.some(exposure => exposure.wasConfirmedExposure && isConfirmedExposureInvalid(exposure))
+    , [exposures]);
+  
+  const disableFlightAddition : boolean= React.useMemo(() => 
+    exposures.some(exposure => exposure.wasAbroad && isFlightInvalid(exposure))
+    , [exposures]);
+
+  const [wereConfirmedExposures, setWereConfirmedExposures] = useState<boolean>(false);
+  const [wereFlights, setWereFlights] = useState<boolean>(false);
+
+  const doesHaveConfirmedExposures = (checkedExposures: Exposure[]) => checkedExposures.some(exposure => exposure.wasConfirmedExposure)
+  const doesHaveFlights = (checkedExposures: Exposure[]) => checkedExposures.some(exposure => exposure.wasAbroad)
+
+  React.useEffect(() => {
+    (wereConfirmedExposures && !doesHaveConfirmedExposures(exposures)) && onExposureAdded(true, false);
+  }, [wereConfirmedExposures])
+
+  React.useEffect(() => {
+    (wereFlights && !doesHaveFlights(exposures)) && onExposureAdded(false, true);
+  }, [wereFlights])
```

Comment:
what is the difference between wereFlights and doesHaveFlights?

Reply:
also maybe a clearer name for wereFlights?

Reply:
doesHaveFlights is a function that checks if there are already flights before the toggle and wereFlights holds the toggle itself

Reply:
i dont really understand the condition in the effect 🤕 
if there were flights, and there arent any now (!doesHaveFlights) then updated the exposures context with wasAbroad true? should it not be false since !doesHaveFlights?
what am i missing, im a bit confused

Reply:
The effect goal is to add default exposure / flight if there arent any so the user will have a default form to begin with. The diffrerence between the conditions is that:
doesHaveFlights == are there ALREADY fligthts on db / state
wereFlights == the toggle status

Reply:
oh ok yeah so its the oppisite than what i thought (WereFlights is current and doesHave is past (db))